docs/user/troubleshooting.md

```diff
@@ -50,3 +54,318 @@ To download the SOS report, run the following command on the device:
+
+```console
+flightctl console device/${device_name} -- journalctl -u flightctl-agent | grep -i -E "(layer|timeout|size)"
+```
+
+Inspect image layers manually (if accessible):
+
+```console
+flightctl console device/${device_name} -- podman manifest inspect registry.example.com/os-image:latest
+```
+
+#### Resolution
+
+1. **Increase available storage**:
+
+   ```console
+   flightctl console device/${device_name} -- podman system prune -a --volumes
+   flightctl console device/${device_name} -- docker system prune -a  # if using Docker
+   ```
+
+   > [!WARNING]
+   > These commands remove all unused images, including downloaded images awaiting deployment.
+
+2. **Adjust resource thresholds** to allow more storage usage during updates:
+
+   ```yaml
+   resources:
+   - monitorType: Disk
+     path: /
+     alertRules:
+     - severity: Critical
+       percentage: 85  # Was 75, increased to allow more space for downloads
+   ```
+
+3. **Use scheduled downloads** during low-usage periods:
+
+   ```yaml
+   updatePolicy:
+     downloadSchedule:
+       at: "0 2 * * *"      # Download at 2 AM when usage is typically low
+       timeZone: "UTC"      # Set explicitly; adjust per fleet locale
+   ```
+
+### Prefetch Status Shows "Buffer Full"
+
+#### Symptoms
+
+- Prefetch operations fail with "buffer full" messages
+- Multiple large images queued simultaneously
+- Agent logs show prefetch scheduling failures
+
+#### Diagnosis
+
+Check the number of concurrent prefetch operations:
+
+```console
+flightctl console device/${device_name} -- journalctl -u flightctl-agent | grep -Ei 'prefetch.*(queue|buffer)'
+```
+
+#### Resolution
+
+1. **Wait for current downloads** to complete before queuing more
+2. **Stagger updates** across devices in a fleet to reduce load
+3. **Monitor and adjust** the agent's pull timeout configuration
+
+### Memory Pressure During Updates
+
+#### Symptoms
+
+- Device becomes unresponsive during OS updates
+- Out of memory errors in system logs
+- Applications killed by OOM killer during updates
+
+#### Diagnosis
+
+Check memory usage and OOM events:
+
+```console
+flightctl console device/${device_name} -- dmesg | grep -Ei 'killed|oom'
+flightctl console device/${device_name} -- free -h
+```
+
+Monitor memory during update process:
+
+```console
+flightctl console device/${device_name} -- watch "free -h && echo '---' && ps aux --sort=-%mem | head -10"
+```
+
+#### Resolution
+
+1. **Configure memory monitoring** with appropriate thresholds:
+
+   ```yaml
+   resources:
+   - monitorType: Memory
+     samplingInterval: 30s
+     alertRules:
+     - severity: Warning
+       duration: 5m
+       percentage: 75
+     - severity: Critical
+       duration: 2m
+       percentage: 85
+   ```
+
+2. **Schedule updates during low-memory periods**:
+
+   ```yaml
+   updatePolicy:
+     updateSchedule:
+       at: "0 3 * * 0"  # Sunday 3 AM when fewer processes are running
+       timeZone: "UTC"
```

Comment:
Just a reminder to include startGraceDuration in all policy examples or this fires at 3:00 exactly and can be missed.